pkg/api/agentPoolOnlyApi/v20180331/merge_test.go

```diff
@@ -163,3 +163,240 @@ func TestMerge_EnableRBAC(t *testing.T) {
 	}
 
 }
+
+func TestMerge_AAD(t *testing.T) {
+	// Partial AAD profile was passed during update
+	newMC := &ManagedCluster{
+		Properties: &Properties{
+			EnableRBAC: helpers.PointerToBool(true),
+			AADProfile: &AADProfile{
+				ClientAppID: "1234-5",
+				ServerAppID: "1a34-5",
+				TenantID:    "c234-5",
+			},
+		},
+	}
+
+	existingMC := &ManagedCluster{
+		Properties: &Properties{
+			DNSPrefix:  "something",
+			EnableRBAC: helpers.PointerToBool(true),
+			AADProfile: &AADProfile{
+				ClientAppID:     "1234-5",
+				ServerAppID:     "1a34-5",
+				ServerAppSecret: "ba34-5",
+				TenantID:        "c234-5",
+			},
+		},
+	}
+
+	e := newMC.Merge(existingMC)
+	if e != nil {
+		t.Error("expect error to be nil")
+	}
+
+	if newMC.Properties.AADProfile == nil {
+		t.Error("AADProfile should not be nil")
+	}
+
+	if newMC.Properties.AADProfile.ServerAppSecret == "" {
+		t.Error("ServerAppSecret should not be empty")
+	}
+
+	if newMC.Properties.AADProfile.ServerAppID != "1a34-5" {
+		t.Error("ServerAppID should not be empty")
```

Comment:
nit: should be "ServerAppID did not have the expected value after merge"

Reply:
Fixed